optimum/exporters/ipex/cache_utils.py

```diff
@@ -38,13 +39,14 @@ def __init__(
         config: PretrainedConfig,
         max_batch_size: int,
         max_cache_len: int,
-        device,
+        device=None,
         dtype=None,
-        layer_device_map=None,
         **kwargs,
     ) -> None:
         super().__init__()
         self.max_batch_size = max_batch_size
+        default_device = torch.device("xpu") if ipex._C._has_xpu() else torch.device("cpu")
```

Comment:
this seems more like a global variable, also if I understand correctly, if xpu device is available and ipex has xpu, it's okay, but what if xpu device is available and current ipex doesn't have xpu support, maybe warning the user and then proceding with cpu device makes sense.